src/crawlers/cheerio_crawler.js

```diff
@@ -552,6 +594,27 @@ class CheerioCrawler {
             }
         });
     }
+
+    /**
+     * Handles blocked request
+     * @param session {Session}
+     * @param statusCode {Number}
+     * @private
+     */
+    _handleBlockedRequest(session, statusCode) {
```

Comment:
This could probably be a method of Session. The `STATUS_CODES_BLOCKED` could be a configuration parameter of Session with defaults that you can override and you'd just call `Session.checkStatus(statusCode)` and it would automatically update (retire/markBad) and throw if the session is unusable.